packages/frontend/src/stdlib/analyses/decapodes.tsx

```diff
@@ -343,7 +362,7 @@ const makeSimulationCode = (data: SimulationData) =>
 
     f = simulator(system.geometry.dualmesh, system.generate, DiagonalHodge());
 
-    soln = run_sim(f, system.init, 50.0, ComponentArray(k=0.5,));
+    soln = run_sim(f, system.init, ${data.duration}, ComponentArray(k=0.5,));
```

Comment:
This works but having to do more code generation/interpolation, minor though it is, is a code smell.

My suggestion would be add a `duration` parameter to `PodeSystem` and then replace this interpolation by the literal `system.duration`.